src/core/MOM_open_boundary.F90

```diff
@@ -6145,6 +6156,56 @@ subroutine rotate_OBC_segment_data(segment_in, segment, turns)
     segment%field(n)%value = segment_in%field(n)%value
   enddo
 
+  if (allocated(segment_in%SSH)) &
+        call rotate_array(segment_in%SSH, turns, segment%SSH)
+  if (allocated(segment_in%cg)) &
+       call rotate_array(segment_in%cg, turns, segment%cg)
+  if (allocated(segment_in%htot)) &
+       call rotate_array(segment_in%htot, turns, segment%htot)
+  if (allocated(segment_in%dztot)) &
+       call rotate_array(segment_in%dztot, turns, segment%dztot)
+  if (allocated(segment_in%h)) &
+       call rotate_array(segment_in%h, turns, segment%h)
+  if (allocated(segment_in%normal_vel)) &
+       call rotate_array(segment_in%normal_vel, turns, segment%normal_vel)
+  if (allocated(segment_in%normal_trans)) &
+       call rotate_array(segment_in%normal_trans, turns, segment%normal_trans)
+  if (allocated(segment_in%normal_vel_bt)) &
+       call rotate_array(segment_in%normal_vel_bt, turns, segment%normal_vel_bt)
+  if (allocated(segment_in%tangential_vel)) &
+       call rotate_array(segment_in%tangential_vel, turns, segment%tangential_vel)
+  if (allocated(segment_in%tangential_grad)) &
+       call rotate_array(segment_in%tangential_grad, turns, segment%tangential_grad)
+  if (allocated(segment_in%grad_normal)) &
+       call rotate_array(segment_in%grad_normal, turns, segment%grad_normal)
+  if (allocated(segment_in%grad_tan)) &
+       call rotate_array(segment_in%grad_tan, turns, segment%grad_tan)
+  if (allocated(segment_in%grad_gradient)) &
+       call rotate_array(segment_in%grad_gradient, turns, segment%grad_gradient)
+  if (allocated(segment_in%rx_norm_rad)) &
```

Comment:
The call on the following line that is going to copy the x- norm_rad array of the input segment into the y- norm_rad of the rotated component should work when there are 1 or 3 turns (90 or 270 degrees), but I don't see how this would be the right thing to do when we are rotating by 2 turns (180 degrees).  The same consideration would apply to lines 6188, 6190 and 6192.